docs/content/troubleshooting/log_status_collect.md

```diff
@@ -1,41 +1,121 @@
+    
+    for pod in $nodepods; kubectl logs -n <namespace> $pod -c $container > /tmp/logs_collection/${pod}_${container}.log;done;done
 
-## Status collection for CSI pods, daemonset, and statefulset
 
-`kubectl get all -n <namespace>  -l csi`
+### Log collection for all CSI controller containers
 
-### Log collection for IBM block storage CSI driver controller
+To collect logs for all pods and containers, use the following commands:
+    
+    for container in kubectl get -n default pod ibm-block-csi-controller-0 -o jsonpath='{.spec.containers[*].name}';do kubectl logs -n default ibm-block-csi-controller-0 -c $container > /tmp/logs_collection/ibm-block-csi-controller-0_${container}.log;done
 
+
+### Log collection for CSI operator logs
+To collect CSI operator logs, use the following commands:
+
+    operatorpod=`kubectl get pods --all-namespaces |grep ibm-block-csi-operator|awk '{print $2}'`
+    kubectl logs $operatorpod -n <namespace> > logs/operator
+
+
+### Collecting details of all CSI objects and components
+
+`kubectl describe all -l product=ibm-block-csi-driver -n <namespace>`
+
+For example:
+
+    kubectl describe all -l csi -n <namespace> > logs/describe_ibm-block-csi-driver
+
+
+### Status collection for CSI pods, daemonset, and statefulset
+`kubectl get all -n <namespace>  -l product=ibm-block-csi-driver`
+
+For example:
+
+    kubectl get all -n <namespace> -l csi > get_all_ibm-block-csi-driver
+
+
+
+### Log collection for the CSI driver controller
 `kubectl log -f -n <namespace> ibm-block-csi-controller-0 -c ibm-block-csi-controller`
 
-### Log collection for IBM block storage CSI driver node (per worker node or PODID)
+For example:
 
+    kubectl log -f -n <namespace> ibm-block-csi-controller-0 -c ibm-block-csi-controller > logs/ibm-block-csi-controller
+
+
+### Log collection for the CSI driver node (per worker node or PODID)
 `kubectl log -f -n <namespace> ibm-block-csi-node-<PODID> -c ibm-block-csi-node`
 
-## Log collection for Operator for IBM block storage CSI driver
+For example:
+    
+    kubectl log -f -n <namespace> ibm-block-csi-node-<PODID> -c ibm-block-csi-node > logs/csi-node-<PODID>
 
-`kubectl log -f -n <namespace> ibm-block-csi-operator-<PODID> -c ibm-block-csi-operator`
 
-## Detecting errors
+### Log collection for unbound PVCs
+`kubectl describe pvc <pvc-name>`
 
-To help pinpoint potential causes for stateful pod failure:
+For example:
 
-1.  Verify that all CSI pods are running.
+    kubectl describe pvc <pvc-name> > logs/pvc_not_bounded
 
-    ```
-    kubectl get pods -n <namespace> -l csi
-    ```
 
-2.  If a pod is not in a _Running_ state, run the following command:
+### Log collection for pods not in the _Running_ state
+`kubectl describe pod <not-running-pod-name>`
 
-    ```
-    kubectl describe -n <namespace> pod/<pod-name>
-    ```
+For example:
 
-    View the logs.
+    kubectl describe pod <pod-name> > logs/pod_not_running
```

Comment:
I don't see here anything specific to our csi, nor any csi.
I don't think we should maintain this

Reply:
if it's not log collection would it be a status collection?
In any case - it was requested by Support and is used by them in cases.  While it may not be csi specific there is value in keeping it here.
@roysahar  please also comment here as we reviewed this together as well.

Reply:
unlike the log, the status is one of the (many) fields in the `describe` command output, so it's better, but still [not accurate](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-application-introspection/#using-kubectl-describe-pod-to-fetch-details-about-pods).
I'm not sure why Support wants this to be specifically in the **our CSI** docs.
if they are not looking for a complete procedure to follow in all cases, then IMO a link to [this](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#viewing-finding-resources) should be more than enough

Reply:
completed per our meeting